File: JavaScript-Solutions/title-case-a-sentence.js

function titleCase(str) {
  var strArray = str.split(" ");
  for (var i = 0; i < strArray.length; i++) {
    strArray[i] = strArray[i].charAt(0).toUpperCase() +                                            
      strArray[i].slice(1).toLowerCase();
  }
  str = strArray.join(" ");
  return str;
}

File: Java-Solutions/title-case-a-sentence.java
package titlecaseasentence;

import java.lang.String;
import java.util.Arrays;

public class TitleCaseASentence {

    public static void main(String[] args) {
        
    }

    public static String titleCaseASentence(String sentence) {
        String[] wordArray = sentence.split(" ");
        for (int i = 0; i < wordArray.length; i++) {
            String firstLetter = wordArray[i].substring(0, 1).toUpperCase();
            String restOfWord = wordArray[i].substring(1).toLowerCase();
            wordArray[i] = firstLetter + restOfWord;
        }
        
        String newSentence = String.join(" ", wordArray);
        return newSentence;
    }

}

